### PR TITLE
[UIKit] Update bindings to Xcode 11 Beta 6

### PR DESCRIPTION
**Pull request by dalexsoto**
Breaking changes are expected, Apple deprecated some methods that we better remove now.

### CODE REVIEW

File: src/uikit.cs [hunk 1]
```diff
@@ -3284,14 +3284,14 @@ interface UIContextMenuInteractionDelegate {
...
 
-		[Export ("contextMenuInteractionWillPresent:")]
-		void WillPresent (UIContextMenuInteraction interaction);
+		[Export ("contextMenuInteraction:willDisplayMenuForConfiguration:animator:")]
+		void WillDisplayMenu (UIContextMenuInteraction interaction, UIContextMenuConfiguration configuration, [NullAllowed] IUIContextMenuInteractionAnimating animator);
 
-		[Export ("contextMenuInteractionDidEnd:")]
-		void DidEnd (UIContextMenuInteraction interaction);
+		[Export ("contextMenuInteraction:willEndForConfiguration:animator:")]
+		void WillEndForConfiguration (UIContextMenuInteraction interaction, UIContextMenuConfiguration configuration, [NullAllowed] IUIContextMenuInteractionAnimating animator);
```
Comment (spouliot) [inline src/uikit.cs]: `ForConfiguration` seems redundant with the parameter. Swift uses overloads for all `contextMenuInteraction` so it's not really a good substitute for the name.

File: src/uikit.cs [hunk 2]
```diff
@@ -3284,14 +3284,14 @@ interface UIContextMenuInteractionDelegate {
 		[return: NullAllowed]
 		UITargetedPreview GetPreviewForDismissingMenu (UIContextMenuInteraction interaction, UIContextMenuConfiguration configuration);
 
-		[Export ("contextMenuInteraction:willCommitWithAnimator:")]
-		void WillCommit (UIContextMenuInteraction interaction, IUIContextMenuInteractionCommitAnimating animator);
+		[Export ("contextMenuInteraction:willPerformPreviewActionForMenuWithConfiguration:animator:")]
+		void WillPerformPreviewActionForMenu (UIContextMenuInteraction interaction, UIContextMenuConfiguration configuration, IUIContextMenuInteractionCommitAnimating animator);
```
Comment (VincentDondain) [inline src/uikit.cs]: Why are we keeping all the `ForXXX` in this type? Isn't that referring to the first argument which is a `UIContextMenuInteraction`.

File: src/uikit.cs (hunk 1 above)
Comment (VincentDondain) [reply src/uikit.cs]: And I'm not quite sure about the other `ForXXX`

Comment (dalexsoto) [reply src/uikit.cs]: Yeah I think plain `WillEnd` is ok

```suggestion
		void WillEnd (UIContextMenuInteraction interaction, UIContextMenuConfiguration configuration, [NullAllowed] IUIContextMenuInteractionAnimating animator);
```

File: src/uikit.cs [hunk 3]
```diff
@@ -4267,8 +4255,8 @@ interface UICollectionViewDelegate : UIScrollViewDelegate {
 		UITargetedPreview GetPreviewForDismissingContextMenu (UICollectionView collectionView, UIContextMenuConfiguration configuration);
 
 		[NoWatch, NoTV, iOS (13,0)]
-		[Export ("collectionView:willCommitMenuWithAnimator:")]
-		void WillCommitMenu (UICollectionView collectionView, IUIContextMenuInteractionCommitAnimating animator);
+		[Export ("collectionView:willPerformPreviewActionForMenuWithConfiguration:animator:")]
+		void WillPerformPreviewActionForMenu (UICollectionView collectionView, UIContextMenuConfiguration configuration, IUIContextMenuInteractionCommitAnimating animator);
```
Comment (VincentDondain) [inline src/uikit.cs]: Not sure the `ForMenu` adds much value here.

Comment (spouliot) [reply src/uikit.cs]: agreed, it's not a direct `UIMenu` argument - but the signature makes it obvious

File: src/uikit.cs (hunk 2 above)
Comment (dalexsoto) [reply src/uikit.cs]: Agreed, it adds very little value

```suggestion
		void WillPerformPreviewAction (UIContextMenuInteraction interaction, UIContextMenuConfiguration configuration, IUIContextMenuInteractionCommitAnimating animator);
```

File: src/uikit.cs (hunk 3 above)
Comment (dalexsoto) [reply src/uikit.cs]: Fixing!

```suggestion
		void WillPerformPreviewAction (UICollectionView collectionView, UIContextMenuConfiguration configuration, IUIContextMenuInteractionCommitAnimating animator);
```

**Comment by dalexsoto** [comment]
Thanks @VincentDondain and @spouliot I addressed your comments

File: src/uikit.cs [hunk 4]
```diff
@@ -12620,8 +12608,8 @@ interface UITableViewSource {
 		UITargetedPreview GetPreviewForDismissingContextMenu (UITableView tableView, UIContextMenuConfiguration configuration);
 
 		[NoWatch, NoTV, iOS (13,0)]
-		[Export ("tableView:willCommitMenuWithAnimator:")]
-		void WillCommitMenu (UITableView tableView, IUIContextMenuInteractionCommitAnimating animator);
+		[Export ("tableView:willPerformPreviewActionForMenuWithConfiguration:animator:")]
+		void WillPerformPreviewActionForMenu (UITableView tableView, UIContextMenuConfiguration configuration, IUIContextMenuInteractionCommitAnimating animator);
```
Comment (dalexsoto) [inline src/uikit.cs]:
```suggestion
		void WillPerformPreviewAction (UITableView tableView, UIContextMenuConfiguration configuration, IUIContextMenuInteractionCommitAnimating animator);
```

Comment (VincentDondain) [inline src/uikit.cs]: @dalexsoto I think you can remove this ForMenu and the one below too

File: src/uikit.cs [hunk 5]
```diff
@@ -13002,8 +12990,8 @@ interface UITableViewDelegate {
...
```
Comment (dalexsoto) [inline src/uikit.cs]:
```suggestion
		void WillPerformPreviewAction (UITableView tableView, UIContextMenuConfiguration configuration, IUIContextMenuInteractionCommitAnimating animator);
```

File: src/uikit.cs [hunk 6]
```diff
@@ -3284,14 +3284,14 @@ interface UIContextMenuInteractionDelegate {
 		[return: NullAllowed]
 		UITargetedPreview GetPreviewForDismissingMenu (UIContextMenuInteraction interaction, UIContextMenuConfiguration configuration);
 
-		[Export ("contextMenuInteraction:willCommitWithAnimator:")]
-		void WillCommit (UIContextMenuInteraction interaction, IUIContextMenuInteractionCommitAnimating animator);
+		[Export ("contextMenuInteraction:willPerformPreviewActionForMenuWithConfiguration:animator:")]
+		void WillPerformPreviewAction (UIContextMenuInteraction interaction, UIContextMenuConfiguration configuration, IUIContextMenuInteractionCommitAnimating animator);
```
Comment (VincentDondain) [inline src/uikit.cs]: @dalexsoto and this animator is not `[NullAllowed]` somehow?

File: src/uikit.cs [hunk 7]
```diff
@@ -12620,8 +12608,8 @@ interface UITableViewSource {
 		UITargetedPreview GetPreviewForDismissingContextMenu (UITableView tableView, UIContextMenuConfiguration configuration);
 
 		[NoWatch, NoTV, iOS (13,0)]
-		[Export ("tableView:willCommitMenuWithAnimator:")]
-		void WillCommitMenu (UITableView tableView, IUIContextMenuInteractionCommitAnimating animator);
+		[Export ("tableView:willPerformPreviewActionForMenuWithConfiguration:animator:")]
+		void WillPerformPreviewAction (UITableView tableView, UIContextMenuConfiguration configuration, IUIContextMenuInteractionCommitAnimating animator);
 
 		// WARNING: If you add more methods here, add them to UITableViewController as well.
```
Comment (VincentDondain) [inline src/uikit.cs]: Is that warning needed? I know you didn't touch it but it's a scary warning

File: src/uikit.cs (hunk 6 above)
Comment (dalexsoto) [reply src/uikit.cs]: Yes, headers do not have `nullable` for that one

```objectivec
- (void)contextMenuInteraction:(UIContextMenuInteraction *)interaction willPerformPreviewActionForMenuWithConfiguration:(UIContextMenuConfiguration *)configuration animator:(id<UIContextMenuInteractionCommitAnimating>)animator;
```

File: src/uikit.cs (hunk 7 above)
Comment (dalexsoto) [reply src/uikit.cs]: Yup, that is needed so we do not forget to do that

File: src/uikit.cs (hunk 6 above)
Comment (VincentDondain) [reply src/uikit.cs]: Could just be missing in the headers but I guess it's minor let's not bother

File: src/uikit.cs (hunk 7 above)
Comment (VincentDondain) [reply src/uikit.cs]: Buuuuuut you didn't do that then

Comment (dalexsoto) [reply src/uikit.cs]: I did but you are right, the WARNING is wrong since what you want is `UITableViewControllerDelegate`

Comment (dalexsoto) [reply src/uikit.cs]:
```suggestion
		// WARNING: If you add more methods here, add them to UITableViewControllerDelegate as well.
```

**Pull request merged**